detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryLetSpec.kt

```diff
@@ -191,6 +191,96 @@ class UnnecessaryLetSpec : Spek({
                 }""")
             assertThat(findings).isEmpty()
         }
+
+        context("destructuring declarations") {
+            it("does not report when destructuring declarations are used more than once 1") {
+                val content = """
+                data class Foo(val a: Int, val b: Int)
+                
+                fun test(foo: Foo) {
+                    foo.let { (a, b) -> a + b }
+                }
+            """
+                val findings = subject.compileAndLint(content)
+                assertThat(findings).isEmpty()
+            }
+
+            it("does not report when destructuring declarations are used more than once 2") {
+                val content = """
+                data class Foo(val a: Int, val b: Int)
+                
+                fun test(foo: Foo) {
+                    foo.let { (a, _) -> a + a }
+                }
+            """
+                val findings = subject.compileAndLint(content)
+                assertThat(findings).isEmpty()
+            }
+
+            it("does not report when destructuring declarations are used more than once 3") {
```

Comment:
Please don't use 2,3,... in test names but try to be a bit more specific.
Given that you have a `context("destructuring declarations")` on top you can probably simplify to 

```suggestion
            it("does not report when destructuring with types") {
```

and so on.